libraries/AP_Scripting/lua_repl.cpp

```diff
@@ -104,26 +104,41 @@ int lua_scripts::pushline(lua_State *L, int firstline) {
     terminal_print(get_prompt(L, firstline));
 
     while (terminal.session) {
-        // reseek to where we need input from, as invalid reads could have done weird stuff, and we want to start from the last valid input
-        int input_fd = AP::FS().open(REPL_IN, O_RDONLY);
-        if (input_fd != -1) {
-            AP::FS().lseek(input_fd, terminal.input_offset, SEEK_SET);
-            ssize_t read_bytes = AP::FS().read(input_fd, buffer, ARRAY_SIZE(buffer) - 1);
-            AP::FS().close(input_fd);
-            if (read_bytes > 0) {
-                // locate the first newline
-                char * newline_chr = strchr(buffer, '\n');
-                if (newline_chr != NULL) {
-                    newline_chr[0] = '\0';
-                    // only advance to the newline
-                    l = strlen(buffer);
-                    terminal.input_offset += l + 1;
+        if (terminal.type == AP_Scripting::relp_type::REPL_FILE)  {
```

Comment:
Its in a while loop so you cant use the a switch and have the breaks still work.

Reply:
You could add a single boolean of "has_complete_line" that you set instead of the break? I'm really looking for switches here, just so that if we add a second option we don't forget to come in here and fix it. If we don't tweak this you'd get a nullptr dereference on the terminal.uart, but it also doesn't seem like it's worth trying to carefully check that here, as it should be valid before we get into the loop.

Reply:
done